pkg/controller/sidecarset/sidecarset_utils.go

```diff
@@ -73,20 +83,21 @@ func (r *ReconcileSidecarSet) updateSidecarSetStatus(sidecarSet *appsv1alpha1.Si
 		return nil
 	}
 
+	sidecarSetClone := sidecarSet.DeepCopy()
 	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		sidecarSet.Status = *status
+		sidecarSetClone.Status = *status
 
-		updateErr := r.Status().Update(context.TODO(), sidecarSet)
+		updateErr := r.Status().Update(context.TODO(), sidecarSetClone)
 		if updateErr == nil {
 			return nil
 		}
 
 		key := types.NamespacedName{
-			Name: sidecarSet.Name,
+			Name: sidecarSetClone.Name,
 		}
 
-		if err := r.Get(context.TODO(), key, sidecarSet); err != nil {
-			klog.Errorf("error getting updated service %s from client", sidecarSet.Name)
+		if err := r.Get(context.TODO(), key, sidecarSetClone); err != nil {
+			klog.Errorf("error getting updated sidecarset %s from client", sidecarSetClone.Name)
```

Comment:
Since input sidecarSet will be used in the later, should we update sidecarSet with the latest version  here?
} else {
    sidecarSet = sidecarSetClone.DeepCopy() 
}